main.go

```diff
@@ -474,8 +486,22 @@ func secureNetwork(testNetwork *network.Network, decelerationFactor float64) {
 }
 
 func startSecurityWorker(peer *network.Peer, pace time.Duration) {
-	for range time.Tick(pace) {
-		sendMessage(peer)
+	if pace == time.Duration(0) {
+		return
+	}
+	ticker := time.NewTicker(pace)
+
+	for {
+		select {
+		case <-ticker.C:
+			if config.IMIF == "poisson" {
+				pace = time.Duration(float64(time.Second) * rand.ExpFloat64() / float64(pace))
```

Comment:
Will the pace be overwritten each time, which makes the lambda keep changing? We might need to use another local var to assign the new time interval.